app/src/main/java/ca/unb/mobiledev/reflexrevolution/GameActivity.java

```diff
@@ -0,0 +1,162 @@
+package ca.unb.mobiledev.reflexrevolution;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.CountDownTimer;
+import android.view.View;
+import android.widget.Button;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+public class GameActivity extends AppCompatActivity {
+    //Time in ms
+    //Could get rid of this and do random intervals or scale it off of score
+    private final int TIME_BETWEEN_LOOPS = 1000;
+
+    private TextView timeText;
+    private TextView scoreText;
+    private LinearLayout layout; //Layout we should add new UI elements to
+
+    private CountDownTimer timer;
+    private String currentInstruction = null;
+    private ArrayList<String> instructions;
+
+    private int timeCount;
+    private int score;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.game);
+
+        timeText = findViewById(R.id.timerText);
+        scoreText = findViewById(R.id.currentScoreText);
+        layout = findViewById(R.id.layout);
+        score = 0;
+
+        createInstructions();
+        updateTimerText();
+        updateScoreText();
+        resetTimer();
+    }
+
+    //Fill the instructions list with instructions
+    private void createInstructions(){
+        instructions = new ArrayList<>();
+
+        //Add instructions here
+        instructions.add("Button"); //Temp instruction for testing
+    }
```

Comment:
Do you think it would also be possible to just pass in a string for "GameMode" from the MainActivity, and then build the list of instructions in GameActivity based on that game mode? Or would it be better to just pass the ArrayList right from MainActivity:

Reply:
Yeah, that could definitely work too. Just passing a string would make it a bit simpler, and whether we keep the lists in the MainActivity or the GameActivity doesn't really make a difference.